src/UhdmAst.cpp

```diff
@@ -331,6 +331,16 @@ AstNode* get_value_as_node(vpiHandle obj_h, bool need_decompile = false) {
             } else {
                 valStr = s;
                 if (valStr.find('\'') == std::string::npos) {
+                    if (vpi_get(vpiSize, obj_h) == -1) {
+                        std::string actualValStr;
+                        if (valStr == "0")
+                            actualValStr = "'0";
+                        else
+                            actualValStr = "'1";
```

Comment:
This is a consequence of Surelog's convention, so check the value here and make sure it's not something unexpected.

Reply:
It will be value after elaboration.  The only thing that comes to my mind is to check if all characters are digits. Do you have a better idea?

Reply:
The check and comment look good, thanks!